### PR TITLE
rsx/nv3089: Fix image_in source line length estimation

### PR DESCRIPTION
**Pull request by elad335**
Fixes https://github.com/RPCS3/rpcs3/issues/12892

in_w was twice the size of out_w, all other parameters are equal incluing in and out pitch. clip_w was equal to out_w. No actual scaling is done.
Seems like we miscalculated the true input line length which caused the crash.

### CODE REVIEW

File: rpcs3/Emu/RSX/NV47/HW/nv3089.cpp [hunk 1]
```diff
@@ -187,19 +187,26 @@ namespace rsx
 			const u32 in_offset = in_x * in_bpp + in_pitch * in_y;
 			const u32 out_offset = out_x * out_bpp + out_pitch * out_y;
 
-			const u32 src_line_length = (in_w * in_bpp);
+			const u32 src_line_length = (std::min(in_w, clip_w) * in_bpp);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: This is wrong. Clip dimensions are on the output, not input. You can easily see how this breaks when you downscale so that in_w is supposed to be larger than clip_w. Cropping is also a valid use of blit engine and there width will be greater than the output clip box size.

Comment (elad335) [reply rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: But realhw did not read the bytes that were clipped away, because the game works on realhw.
I guess I should add a check for the scale_x parameter as well? it's 1.0 for the transfer.

Comment (elad335) [reply rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: Fixed

Comment (kd-11) [reply rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: You're correct in that sense - it did not read specifically because it did not need to, not because it's a rule. I'm avoiding adding a nest of ifs to try and dodge around edge cases.  Adding the scale_x = 1 test will fix this one game but what if its not 1? I believe the range checking here is premature since the input and output sizes will depend greatly on the final scaling or copy path taken.
For input it is actually the inverse scaled image dimensions from the output plus in pitch taken into account.
I believe the in_pitch here should help figure out the real in_w. What is the in_pitch? Because it doesn't matter if you clip or not, the pitch will still be part of the 2D calculation unless the height is 1.

Comment (elad335) [reply rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: in_pitch and out_picth both are 32 bytes. in_w = 16, out_w/clip_w = 8 and in_bpp/out_bpp both are 4. the overestimation is 32 bytes because 16 * 4 = 64 where it should be 32 bytes.

Comment (kd-11) [reply rpcs3/Emu/RSX/NV47/HW/nv3089.cpp]: Then you missed the more general solution there. in_pitch/in_bpp = 8, not 16. The request is misconfigured.
This would fix all variations regardless of scale factor.
```C++
const auto in_width_min = (clip_h > 1 && in_pitch > 0) ? (in_pitch / in_bpp) : in_w;
// Then clamp the src width used in the calculation
```
